thsvs_simpler_vulkan_synchronization.h

```diff
@@ -849,15 +848,15 @@ void thsvsGetVulkanImageMemoryBarrier(
                     break;
             }
 
+            pVkBarrier->oldLayout = layout;
 #ifdef THSVS_ERROR_CHECK_MIXED_IMAGE_LAYOUT
             assert(pVkBarrier->oldLayout == VK_IMAGE_LAYOUT_UNDEFINED ||
                    pVkBarrier->oldLayout == layout);
 #endif
-            pVkBarrier->oldLayout = layout;

```

Comment:
This change breaks the point of the THSVS_ERROR_CHECK_MIXED_IMAGE_LAYOUT, but obviously it wasn't working in the first place - fixing it probably just needs pVkBarrier->oldLayout to be initialised to VK_IMAGE_LAYOUT_UNDEFINED.